packages/server/tests/clients/servicesClient.ts

```diff
@@ -0,0 +1,246 @@
+            .setTokenName(`relay-acceptance token ${symbol}`)
+            .setTokenSymbol(symbol)
+            .setDecimals(3)
+            .setInitialSupply(new Hbar(1000).toTinybars())
+            .setTreasuryAccountId(this._thisAccountId()));
+
+        this.logger.trace(`get token id from receipt`);
+        const tokenId = resp?.tokenId;
+        this.logger.info(`token id = ${tokenId?.toString()}`);
+        return tokenId;
+    };
+
+    async associateToken(tokenId) {
+        await this.executeAndGetTransactionReceipt(
+            await new TokenAssociateTransaction()
+                .setAccountId(this._thisAccountId())
+                .setTokenIds([tokenId]));
+
+        this.logger.debug(
+            `Associated account ${this._thisAccountId()} with token ${tokenId.toString()}`
+        );
+    }
+
+    async transferToken(tokenId, recipient: AccountId, amount= 10) {
+        await this.executeAndGetTransactionReceipt(new TransferTransaction()
+            .addTokenTransfer(tokenId, this._thisAccountId(), -amount)
+            .addTokenTransfer(tokenId, recipient, amount));
+
+        this.logger.debug(
+            `Sent 10 tokens from account ${this._thisAccountId()} to account ${recipient.toString()} on token ${tokenId.toString()}`
+        );
+
+        const balances = await this.executeQuery(new AccountBalanceQuery()
+            .setAccountId(recipient));
+
+        this.logger.debug(
+            `Token balances for ${recipient.toString()} are ${balances.tokens
+                .toString()
+                .toString()}`
+        );
+    }
+
+    async createParentContract(contractJson) {
+        const contractByteCode = (contractJson.deployedBytecode.replace('0x', ''));
+
+        const fileReceipt = await this.executeAndGetTransactionReceipt(new FileCreateTransaction()
+            .setKeys([this.client.operatorPublicKey || this.DEFAULT_KEY])
+            .setContents(contractByteCode));
+
+        // Fetch the receipt for transaction that created the file
+        // The file ID is located on the transaction receipt
+        const fileId = fileReceipt?.fileId;
+        this.logger.info(`contract bytecode file: ${fileId?.toString()}`);
+
+        // Create the contract
+        const contractReceipt = await this.executeAndGetTransactionReceipt(new ContractCreateTransaction()
+            .setConstructorParameters(
+                new ContractFunctionParameters()
+            )
+            .setGas(75000)
+            .setInitialBalance(100)
+            .setBytecodeFileId(fileId || "")
+            .setAdminKey(this.client.operatorPublicKey || this.DEFAULT_KEY));
+
+        // The contract ID is located on the transaction receipt
+        const contractId = contractReceipt?.contractId;
+
+        this.logger.info(`new contract ID: ${contractId?.toString()}`);
+
+        return contractId;
+    };
+
+    async executeContractCall(contractId) {
```

Comment:
I would prefer passing in the functionName and `ContractFunctionParameters` so this method can be reused

Reply:
updated it although I think that we should migrate to using the `ethers.Contract` instance for other types of tests.